lib/iterators.js

```diff
@@ -55,25 +61,38 @@ const SchemaTypesToIterate = Object.freeze({
  * @param {TraverseSchemas} callback 
  * @param {SchemaTypesToIterate[]} schemaTypesToIterate
  */
+/* eslint-disable sonarjs/cognitive-complexity */
```

Comment:
disabled cognitive-complexity for this one, then too sonar cloud code smell issue is coming.

Reply:
It is fine, the reviewer will check whether it makes sense to disable it. Right of first glance, I don't see how you can reduce the size here, another reviewer might be able to 🙂 

Just a general rule of thumb think about whether it makes sense to refactor the complexity or not. Would it be more complex to split any of the functionality up? If the answer is yes, then disable the eslint, otherwise, it should of course be refactored.

Did that clarify it?

Reply:
Yeah, that makes sense. Also, the PR is ready from my side.